sources/api/schnauzer/src/helpers.rs

```diff
@@ -1301,6 +1366,178 @@ pub fn etc_hosts_entries(
     Ok(())
 }
 
+/// This helper writes out the default OCI runtime spec.
+///
+/// Default settings are specified in the following files:
+/// * sources/models/shared-defaults/oci-resource-limits.toml
+/// * sources/models/shared-defaults/oci-capabilities.toml
+
+pub fn oci_defaults(
+    helper: &Helper<'_, '_>,
+    _: &Handlebars,
+    _: &Context,
+    renderctx: &mut RenderContext<'_, '_>,
+    out: &mut dyn Output,
+) -> Result<(), RenderError> {
+    // To give context to our errors, get the template name (e.g. what file we are rendering), if available.
+    trace!("Starting oci_defaults helper");
+    let template_name = template_name(renderctx);
+    trace!("Template name: {}", &template_name);
+
+    // Check number of parameters, must be exactly two (OCI spec section to render and settings values for the section)
+    trace!("Number of params: {}", helper.params().len());
+    check_param_count(helper, template_name, 2)?;
+    trace!("params: {:?}", helper.params());
+
+    // Validate the values given to the template
+    trace!("Getting the desired OCI spec section to render in THIS call from the first param");
+    let current_oci_spec_section = get_param(helper, 0)?;
+    let current_oci_spec_section_was_null: bool = current_oci_spec_section.is_null();
+    trace!(
+        "OCI Spec section to render was null? {}",
+        current_oci_spec_section_was_null
+    );
+
+    trace!(
+        "Matching whether the desired OCI spec section is supported ({})",
+        current_oci_spec_section
+            .as_str()
+            .with_context(|| error::InvalidTemplateValueSnafu {
+                expected: SUPPORTED_OCI_SPEC_SECTIONS.join(", "),
+                value: current_oci_spec_section.to_owned(),
+                template: template_name.to_owned(),
+            })?
+    );
+    let good_oci_spec_section = match current_oci_spec_section {
+        Value::String(current_oci_spec_section)
+            if (SUPPORTED_OCI_SPEC_SECTIONS.contains(&current_oci_spec_section.as_str())) =>
+        {
+            current_oci_spec_section.as_str()
+        }
+        &handlebars::JsonValue::String(_)
+        | Value::Null
+        | Value::Bool(_)
+        | Value::Number(_)
+        | Value::Array(_)
+        | Value::Object(_) => {
+            trace!(
+                "Error condition hit! Bad path taken: current OCI spec section type is unexpected: {:?}",
+                current_oci_spec_section.type_id()
+            );
+            return Err(RenderError::from(
+                error::TemplateHelperError::InvalidTemplateValue {
+                    expected: format!("one of: '{}'", SUPPORTED_OCI_SPEC_SECTIONS.join(", ")),
```

Comment:
make this a fully static string: "one of: abc, xyz" and undo the .to_string up above